### DISCUSSION TITLE
Not able to iterate through elements using find() where it works with find elements

### DISCUSSION THREAD

**Discussion by DudeParaK** [ElSnoMan/pyleniumio#212] [Q&A]
Elements = py.find("ul[id='ulid']>li")
    for el in Elements:
        if el.text == 'text':
            el.click()
            break

**Comment by ElSnoMan** [comment]
> 💡TL;DR: I trust that your selector is good, so I won't focus on that. Look at the code block below for a solution to your query


- In Python, variables should be `snake_case`, meaning that they are lowercase and separated by underscores. Titled words, like you did for `Elements`, are meant for things like classes.

- You don't have the proper indentation. I'm assuming that's an error with how you copy and pasted the code, but Python requires spaces to signify code blocks.

- In Pylenium, you get the text of an element using `.text()` since it's a function and not a property

```python
elements = py.find("ul[id='ulid'] > li")
for el in elements:
    if el.text() == "text":
        el.click()
        break
```

**Reply by DudeParaK** [reply]
Hi Carlos
Thanks for the reply
I have tried below as in the below image
  ![Capture](https://user-images.githubusercontent.com/91080081/134296922-3f57c045-3160-46c7-8e75-ca6ad7c013fd.JPG)

**Reply by ElSnoMan** [reply]
Are you able to share the website so we can try on our end? Or can you provide a public website example where you also see this issue so we can debug? This is pretty confusing because `py.find()` resolves to `webdriver.find_elements()`. Without any other context I'm not sure what to tell you. We have many people using this for a huge spread of applications and we have tests around the framework as well and _find()_ is definitely working 😅

Here is the method in `pylenium.driver`. Observe how it resolves to `.find_elements()`

```python
    def find(self, css: str, timeout: int = None) -> Elements:
        """Finds all DOM elements that match the `css` selector.
        * If timeout=None (default), use the default wait_time.
        * If timeout > 0, override the default wait_time.
        * If timeout=0, poll the DOM immediately without any waiting.
        Args:
            css: The selector to use.
            timeout: The number of seconds to wait for this to succeed. Overrides the default wait_time.
        Returns:
            A list of the found elements.
        """
        by = By.CSS_SELECTOR
        self.log.info(f'[STEP] py.find() - Find elements with css: ``{css}``')

        try:
            if timeout == 0:
                elements = self.webdriver.find_elements(by, css)
            else:
                elements = self.wait(timeout).until(
                    lambda x: x.find_elements(by, css), f'Could not find any elements with the CSS ``{css}``'
                )
        except TimeoutException:
            elements = []
        return Elements(self, elements, locator=(by, CSS))
```

**Reply by DudeParaK** [reply]
Thanks for the reply Carlos..how ever I managed with children() and worked well
I have few queries that probably you can answer as I just go through the document 

1.Where and how are the drivers(chrome,firefox..etc) configured corresponding to the browser version in the user's system if I provide that in the json file

2.Did not find the "pylog_level": "info" in the json file,so I added the key and value,Should I type py.log.log("message") to get the logs in the test_result folder as .txt

3.where the should() find the wait time,is it configurable somewhere?
